python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -323,11 +345,30 @@ def run(self, inputWarps, skyInfo, **kwargs):
 
                 coadd_inputs = coadd_inputs_gc[cellInfo.index]
                 self.input_recorder.addVisitToCoadd(coadd_inputs, warp[bbox], weight)
+                if True:
+                    ccd_table = warp.getInfo().getCoaddInputs().ccds.subsetContaining(cell_centers_sky[cellInfo.index])
+                    assert len(ccd_table) > 0, "No CCD from a warp found within a cell."
+                    assert len(ccd_table) == 1, "More than one CCD from a warp found within a cell."
+                    ccd_row = ccd_table[0]
+                else:
+                    for ccd_row in warp.getInfo().getCoaddInputs().ccds:
+                        if ccd_row.contains(cell_centers_sky[cellInfo.index]):
+                            break
+
```

Comment:
> Make a Numpy array of cell centers in sky coordinates outside the loop over warps.

OK, we do this now but as a `dict` instead of an array.

> Compare those detector-coordinate positions to the detector bounding boxes.

We're going to have to do comparisons with polygons, since detectors aren't within rectangles anymore. I'll have to check that those comparisons are efficient as well.

But I'm going to create another ticket to improve the efficiency, since we are still very much limited by PSF evaluation per-cell.

Reply:
> since detectors aren't within rectangles anymore.

That's why I suggested going all the way back to detector coordinates.  But doing this in polygons on the sky or in coadd coordinates may well be more efficient if it avoids the reverse-transform to check for bad extrapolation.

Reply:
Ah, I misunderstood the coordinate system in your comment at first.